python/tank/authentication/user_impl.py

```diff
@@ -206,7 +206,19 @@ def __init__(self, host, login, session_token, http_proxy, password=None, cookie
         self._login = login
         self._session_token = session_token
         self._cookies = cookies
-        self._saml_expiration = saml_expiration
+
+        # saml expiration is meaningful only with SSO enabled.
+        # The presence of cookies implies potential SSO usage.
+        if self._cookies is None:
+            self._saml_expiration = None
+        elif saml_expiration is None:
+            # For lack of accurate information, we will indicates that the
+            # credentials are expired  by setting the expiration to a moment
+            # in the past.
+            # @TODO: obtain the expiration from the cookies themselves, if possible.
```

Comment:
seems we should tighten this up by saying `if saml_flag exists in cookies then...` rather than just looking if we have cookies. I don't have the full picture, but if i understand correctly this seems a little brittle. What if there is other stuff that begins to depend on cookies?